testing/cli-e2etest/testscenarios/demo/list_demos_test.go

```diff
@@ -108,18 +108,19 @@ func TestListDemos(t *testing.T) {
 		result := tracetestcli.Exec(t, "list demo --sortBy name --sortDirection asc --output json", tracetestcli.WithCLIConfig(cliConfig))
 		helpers.RequireExitCodeEqual(t, result, 0)
 
-		demos := helpers.UnmarshalJSON[[]types.DemoResource](t, result.StdOut)
-		require.Len(demos, 2)
+		demos := helpers.UnmarshalJSON[types.ResourceList[types.DemoResource]](t, result.StdOut)
```

Comment:
this scenario was incorrect (as well as the old implementation). it was returning an array of resources instead of a `ResourceList`